extension-style/src/main/java/com/mapbox/maps/extension/style/sources/Source.kt

```diff
@@ -48,13 +50,31 @@ abstract class Source(
 
   internal var delegate: StyleManagerInterface? = null
 
+  private var styleObjectIsValidMethod: Method? = null
+
   /**
    * Add the source to the Style.
    *
    * @param delegate The style delegate
    */
   override fun bindTo(delegate: StyleInterface) {
     this.delegate = delegate
+    // geojson source is the only one holding async logic -
+    // in order to avoid accessing native style object when MapView is destroyed we need to check
+    // if delegate is still valid.
+    // Sadly due to current architecture it's now easily achievable with reflection only
+    // as this module does not have dependency on main SDK module and Style object
```

Comment:
why don't we add isValid to the StyleInterface? I think it's an acceptable addition as it is a bug fix, and using reflection is not efficient or safe.. And StyleInterface isn't supposed to be implemented by user at all..

Reply:
my thinking was that it's still a public interface and adding new property / method there is still a breaking change. If we're OK with actually extending `StyleInterface` with a new property - I'm 2 hands for it.

Reply:
I think it's fine in this case, as user should not implement the StyleInterface besides writing some tests.. Would like thoughts from the Android team as well @mapbox/maps-android

Reply:
guess we decided to go with that solution